compiler/rustc_typeck/src/impl_wf_check.rs

```diff
@@ -207,7 +207,15 @@ fn enforce_impl_items_are_distinct(tcx: TyCtxt<'_>, impl_def_id: LocalDefId) {
         };
         let span = tcx.def_span(impl_item_ref);
         let ident = impl_item.ident(tcx);
-        match seen_items.entry(ident.normalize_to_macros_2_0()) {
+
+        // For trait impls, we check that there is a single implementation for each trait item.
+        // For inherent impls, we check that there is no two items with the same name.
```

Comment:
Might be worth saying explicitly that "name" here includes the hygiene disambiguator or whatever it is called.

Reply:
```suggestion
        // For inherent impls, we check that there is no two items with the same name, including hygiene.
```